### PR TITLE
Update Lightning Lite docs (4/n)

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Adds missing callback documentation

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source-pytorch/fabric/fabric.rst [hunk 1]
```diff
@@ -195,13 +195,12 @@ We recommend you to convert to :doc:`Lightning <../starter/introduction>`, so yo
...
 ----------
 
-************
-Fabric Flags
-************
+********************
+Fabric Configuration
+********************
 
 Fabric is specialized in accelerated distributed training and inference. It offers you convenient ways to configure
```
Comment (edenlightning) [inline docs/source-pytorch/fabric/fabric.rst:202]:
```suggestion
Fabric is designed to accelerate distributed training and inference. It makes it easy to configure your device and communication strategy, and and to switch seamlessly from one to the other.```

File: docs/source-pytorch/fabric/fabric.rst [hunk 2]
```diff
@@ -389,6 +353,33 @@ To define your own behavior, subclass the relevant class and pass it in. Here's
     fabric = Fabric(plugins=[MyCluster()], ...)
 
 
+callbacks
+=========
+
+The callback system allows you to inject logic into your training loop from an external callback class.
+A callback class is a collection of methods that the training loop can call at a specific point, for example, at the end of an epoch:
```
Comment (edenlightning) [inline docs/source-pytorch/fabric/fabric.rst:360]:
```suggestion
A callback class is a collection of methods that the training loop can call at a specific point in time, for example, at the end of an epoch.
Add callbacks to Fabric to inject logic into your training loop from an external callback class.
```

**Review by edenlightning — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: docs/source-pytorch/fabric/fabric.rst [hunk 3]
```diff
@@ -589,3 +579,19 @@ For single-device strategies, it is a no-op. There are strategies that don't sup
...
+
+Use this to run all registered callback hooks with a given name and inputs.
+It is useful when building a Trainer that allows the user to run arbitrary code at fixed points in the training loop.
+
+.. code-block:: python
+
+    fabric = Fabric(callbacks=[MyCallback()]))
+
+    # Call any hook by name
+    fabric.call("on_train_epoch_end", results={...})
```
Comment (justusschock) [inline docs/source-pytorch/fabric/fabric.rst:595]: here it is not clear, why you pass in `results`. IMO you should include a dummy callback that at least shows the signature for `on_train_epoch_end` and mention this specifically. Or at least link to the callbacks section for more details

File: docs/source-pytorch/fabric/fabric.rst [hunk 4]
```diff
@@ -389,6 +352,33 @@ To define your own behavior, subclass the relevant class and pass it in. Here's
...
+
+    fabric = Fabric(callbacks=[MyCallback()]))
+
+
+Then, in your training loop, you can call a hook by its name. Any callback objects that have this hook will execute it:
+
+.. code-block:: python
+
+    # Call any hook by name
+    fabric.call("on_train_epoch_end", results={...})
```
Comment (justusschock) [inline docs/source-pytorch/fabric/fabric.rst:379]: That means that all callbacks ideally support `**kwargs` to store undesired arguments for other callbacks right? Can we also pass positional arguments or just keyword args?

**Review by justusschock — APPROVED** [review]

File: docs/source-pytorch/fabric/fabric.rst (hunk 4 above)
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst:379]: Right now, all callbacks need to consistently accept the same args. That restriction is probably reasonable. We don't have special handling of args/kwargs right now

File: docs/source-pytorch/fabric/fabric.rst (hunk 3 above)
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst]: Addressed!

File: docs/source-pytorch/fabric/fabric.rst (hunk 4 above)
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst:379]: In the doc I just added

"Pass in additional arguments that the hook requires"

Which should make it clear that if they are required, you need to pass them in. This implies your *args **kwargs handling needs to behave accordingly. 

We may want to inspect signatures and improve error messaging in the future

**Pull request merged**